buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -248,7 +250,20 @@ public static String gitRevision(File rootDir) {
             }
             final String ref = readFirstLine(head);
             if (ref.startsWith("ref:")) {
-                revision = readFirstLine(gitDir.resolve(ref.substring("ref:".length()).trim()));
+                String branchname = ref.substring("ref:".length()).trim();
+                Path refFile = gitDir.resolve(branchname);
+                if (Files.exists(refFile)) {
+                    revision = readFirstLine(refFile);
+                } else {
+                    // Check packed references for commit ID
+                    Pattern p = Pattern.compile("^([a-f0-9]+) " + branchname + "$");
```

Comment:
We could make this even more strict since we know a sha1 hash is 40 characters with `[a-f0-9]{40}`.

Reply:
Also, do we know what the entries that look like `^6d5d330548f66c1ba6d2913737ba93c5d5a39de9` mean? I couldn't find any documentation describing the actual format of this file.

Reply:
Looking at the last lines of my `.git/packed-refs` file…
```
d7d5b3fbc408e3b4060d9df33dec9c43e29e9cbb refs/tags/v7.4.0
^22e1767283e61a198cb4db791ea66e3f11ab9910
```
It looks like we get that carrot notation when Elasticmachine tags a release. I'm not sure what the specific terminology is, but `22e17672` is the commit that's tagged for 7.4 and `git show d7d5b3fbc` shows the tag. Perhaps the first line represents the "annotated tag object"? I wouldn't want to try to get any more definitive without finding the format documentation.

Reply:
Just curious mostly. Seems we can safely ignore those entries.